crates/bevy_pbr/src/material.rs

```diff
@@ -1,321 +1,312 @@
-use crate::{AlphaMode, PbrPipeline, StandardMaterialFlags};
+use crate::{
+    AlphaMode, DrawMesh, MeshPipeline, MeshPipelineKey, MeshUniform, SetMeshBindGroup,
+    SetMeshViewBindGroup,
+};
 use bevy_app::{App, Plugin};
-use bevy_asset::{AddAsset, Handle, HandleUntyped};
-use bevy_ecs::system::{lifetimeless::SRes, SystemParamItem};
-use bevy_math::Vec4;
-use bevy_reflect::TypeUuid;
+use bevy_asset::{AddAsset, Asset, AssetServer, Handle};
+use bevy_core_pipeline::{AlphaMask3d, Opaque3d, Transparent3d};
+use bevy_ecs::{
+    entity::Entity,
+    prelude::World,
+    system::{
+        lifetimeless::{Read, SQuery, SRes},
+        Query, Res, ResMut, SystemParamItem,
+    },
+    world::FromWorld,
+};
 use bevy_render::{
-    color::Color,
-    render_asset::{PrepareAssetError, RenderAsset, RenderAssetPlugin, RenderAssets},
+    mesh::Mesh,
+    render_asset::{RenderAsset, RenderAssetPlugin, RenderAssets},
+    render_component::ExtractComponentPlugin,
+    render_phase::{
+        AddRenderCommand, DrawFunctions, EntityRenderCommand, RenderCommandResult, RenderPhase,
+        SetItemPipeline, TrackedRenderPass,
+    },
     render_resource::{
-        std140::{AsStd140, Std140},
-        BindGroup, Buffer, BufferInitDescriptor, BufferUsages,
+        BindGroup, BindGroupLayout, RenderPipelineCache, RenderPipelineDescriptor, Shader,
+        SpecializedPipeline, SpecializedPipelines,
     },
     renderer::RenderDevice,
-    texture::Image,
+    view::{ExtractedView, Msaa, VisibleEntities},
+    RenderApp, RenderStage,
 };
-use wgpu::{BindGroupDescriptor, BindGroupEntry, BindingResource};
+use std::hash::Hash;
+use std::marker::PhantomData;
+
+pub trait Material: Asset + RenderAsset {
```

Comment:
Basic docs here would be nice; this is an important type that gets linked to.

Reply:
Added!